include/nuttx/addrenv.h

```diff
@@ -193,7 +193,9 @@
 #  define ARCH_SHM_MAXPAGES   (CONFIG_ARCH_SHM_NPAGES * CONFIG_ARCH_SHM_MAXREGIONS)
 #  define ARCH_SHM_REGIONSIZE (CONFIG_ARCH_SHM_NPAGES * CONFIG_MM_PGSIZE)
 #  define ARCH_SHM_SIZE       (CONFIG_ARCH_SHM_MAXREGIONS * ARCH_SHM_REGIONSIZE)
```

Comment:
why not rename SHM_xxx to general name directly, but add UDEV_xxx?

Reply:
By adding one UDEV_NPAGES option, we can keep most SHM_xxx unchanged, and `device mapping` and `SHM` can be two indenpendent choices, this sounds more flexible. There is no other `UDEV_xxx` stuffs needed.

Reply:
@pussuw what's your opinion?

Reply:
Changing the name of the SHM region is unnecessary, what would be a better name. Doing so would also create a regression for downstream projects, so this is not preferred by me.

Why not just use CONFIG_ARCH_SHM_NPAGES for the UDEV/UMAP area as well ? There is no distinction between the two from memory segmentation point of view. The SHM area has a confusing name I agree, but it is really just a general "memory mappings region". Anything that is mapped by mmap can go here; shm objects, shared libraries, udev mappings, whatever. Here is a good description of a process's memory map: https://i.stack.imgur.com/epGfE.png.
Semantically, the SHM region in that figure is the "Memory Mapping Region".

You define `# define ARCH_SHM_VEND (CONFIG_ARCH_SHM_VBASE + ARCH_UMAP_SIZE - 1)` which proves that the UMAP region is just a region of SHM. If your goal is to remove naming confusion, this does not help in my opinion.

Reply:
> 
> You define `# define ARCH_SHM_VEND (CONFIG_ARCH_SHM_VBASE + ARCH_UMAP_SIZE - 1)` which proves that the UMAP region is just a region of SHM. If your goal is to remove naming confusion, this does not help in my opinion.

This will be `ARCH_UMAP_VEND` later if we agree that `UMAP` region is composed of `SHM` and `UDEV`.

To reuse existing SHM configs, my confusion comes from the ARCH_SHM_MAXREGIONS thing as it affects the total virtual region size. Can we remove it from the ARCH_SHM_VEND calculations and simply let SHM_NPAGES defines the size?

Reply:
ARCH_SHM_MAXREGIONS is SysV shared memory stuff. I'm not sure what to do about that. I'm 100% sure no one uses that feature, it is obsolete and a better alternative (shmfs) exists.

Reply:
Yes, SysV shared memory is better to implement on top of shmfs and remove all related config, as I suggest before.

Reply:
@xiaoxiang781216  and @pussuw

Okay, I will drop UDEV_NPAGES and leave SHM stuffs untouched for now. Maybe later we can use mail list or an issue to discuess about how to fix the confusions of SHM stuffs.